src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -9,267 +9,170 @@ foam.CLASS({
       margin: 0 auto;
     }
-    ^ .title-top {
-      font-size: 2.5em;
-      padding-top: 2vh;
-      font-weight: bold;
-      text-align: center;
-      margin-bottom: 0
+    ^content {
+      max-width: 30vw;
+      margin: 4vh auto 0 auto;
+    }
+    ^section {
+    }
+    /* title  */
+    ^ ^section h2 {
+      margin-top: 0;
+      margin-bottom: 4vh;
+      font-size: 2.5rem;
     }
-    ^ .subtitle {
-      color: #525455;
-      font-size: 1em;
+    /* subtitle */
+    ^ ^section .subtitle {
+      color: /*%GREY2%*/ #8e9090;
+      font-size: 1rem;
+      margin-top: 0;
+      margin-bottom: 3vh;
       line-height: 1.5;
-      margin-bottom: 2vh;
       text-align: center;
     }
-    ^ .contents {
-      max-width: 25vw;
-      margin: 0 auto;
-    }
-    ^ .submitBtn {
+    ^link {
+      color: /*%PRIMARY3%*/ #604aff;
+      cursor: pointer;
       text-align: center;
+      padding-top: 1.5vh;
     }
   `,
 
-  sections: [
-    {
-      name: 'resetPasswordSection',
-      title: '',
-      help: 'Create a new password for your account',
-    }
-  ],
-
   properties: [
     {
       class: 'Boolean',
-      name: 'horizontal',
-      documentation: `This property toggles the view from a centered vertical view to a
-      horizontally property display view.`,
-      hidden: true
-    },
-    {
-      class: 'Boolean',
-      name: 'topBarShow',
+      name: 'showHeader',
       documentation: `This property toggles the view from having a top bar displayed.`,
       hidden: true,
-      value: true
-    },
-    {
-      class: 'String',
-      name: 'token',
-      documentation: `This property toggles the view from updating a user password to resetting a user password.`,
       factory: function() {
-        var searchParams = new URLSearchParams(location.search);
-        return searchParams.get('token');
-      },
-      hidden: true
+        // implement a logic to hide logo in certain places
+        return true;
+      }
     },
     {
-      class: 'Password',
-      name: 'originalPassword',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      visibility: function(token) {
-        return ! token ? foam.u2.DisplayMode.RW : foam.u2.DisplayMode.HIDDEN;
+      class: 'Boolean',
+      name: 'isHorizontal',
+      documentation: 'setting this to true makes input fields in passwordModel to be displayed horizontally',
+      factory: function() {
+        return this.mode === 'update';
       }
     },
     {
-      class: 'Password',
-      name: 'newPassword',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      minLength: 6
+      class: 'String',
+      name: 'mode',
+      documentation: 'name of password model used for this view',
+      factory: function() {
+        if ( window.location.hash.includes('personal-settings') ||
```

Comment:
Why do we do this?

Reply:
I am using the url of page the user is currently viewing to determine which password model to use. For instance, the url for updating password page would be https://ablii/personal-setting, and the url for sending password page would be https://abli-domain.com.

Reply:
If you are asking why we need 'mode' property, then I am using it to determine which password model to use for this view. I could've just done everything inside 'passwordModel' property, but it looked more intuitive and cleaner.

Reply:
I'd prefer if we pass in the model then have the changePasswordView define what model to use - especially if it's checking a url hash - which limits our reusability. Let's say for example I wanted to use UpdatePassword outside of the personal-settings, I wouldn't be able to do that without extending this class and rewriting this factory function.

I think having something like: 
{ class: 'foam.nanos.auth.ChangePasswordView', passwordModel: 'foam.nanos.auth.UpdatePassword' }
Would make things a bit easier and flexible to use.